tests/aws_iot_secure_tunneling_client_test.c

```diff
@@ -98,6 +132,11 @@ int main(int argc, char **argv) {
     ASSERT_SUCCESS(aws_condition_variable_wait(&condition_variable, &mutex));
     aws_mutex_unlock(&mutex);
 
+    if (local_proxy_mode == AWS_SECURE_TUNNELING_DESTINATION_MODE) {
```

Comment:
Prefer to use a condition variable to wait here, and trigger it when data is received.

Reply:
I could, but then I will only get one data package and the test client would quit. I am using this test client to print out whatever I get from the other side.